### PR TITLE
[DOCS] Network community ID processor for ingest pipelines

### PR DESCRIPTION
**Pull request by danhermann**
Depends on #66534.

### CODE REVIEW

File: docs/reference/ingest/processors/community-id.asciidoc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+|======
+| Name               | Required | Default       | Description
+| `source_ip`        | no       | `source.ip`   | Field containing the source IP address.
+| `source_port`      | no       | `source.port` | Field containing the source port.
+| `destination_ip`   | no       | `destination.ip` | Field containing the destination IP address.
+| `destination_port` | no       | `destination.port` | Field containing the destination port.
+| `iana_number`      | no       | `network.iana_number` | Field containing the IANA number.
+| `icmp_type`        | no       | `icmp.type`   | Field containing the ICMP type.
+| `icmp_code`        | no       | `icmp.code`   | Field containing the ICMP code.
+| `transport`        | no       | `network.transport` | Field containing the transport protocol.
```
Comment (andrewkroh) [inline docs/reference/ingest/processors/community-id.asciidoc:24]: Should we mention that this field is only used when the iana_number field does not exist?

File: docs/reference/ingest/processors/community-id.asciidoc [hunk 2]
```diff
@@ -0,0 +1,91 @@
+[role="xpack"]
+[testenv="basic"]
+[[community-id-processor]]
+=== Community ID processor
+++++
+<titleabbrev>Community ID</titleabbrev>
+++++
+
+Computes the Community ID for flow data as defined in the
```
[the remark this answers is not in the document]
Comment (danhermann) [reply docs/reference/ingest/processors/community-id.asciidoc:9]: I'll defer to the ES Docs team reviewer on this one as I have no strong inclination either way.

File: docs/reference/ingest/processors/community-id.asciidoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+https://github.com/corelight/community-id-spec[Community ID Specification].
```
Comment (jrodewig) [inline docs/reference/ingest/processors/community-id.asciidoc:10]: Lowercases `community ID`. Also adds a few additional sentences as noted above.
```suggestion
Computes the community ID for network flow data as defined in the
https://github.com/corelight/community-id-spec[Community ID Specification]. You
can use a community ID to correlate network events related to a single flow.

The community ID processor reads network flow data from related
{ecs-ref}[Elastic Common Schema (ECS)] fields by default. If you use the ECS, no
configuration is required.
```

File: docs/reference/ingest/processors/community-id.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/ingest/processors/community-id.asciidoc:9]: I don't think we should copy the paragraphs verbatim. That text is pretty Beats-centric. However, we can include a brief sentence about how a community ID is used and the ECS defaults. I've included a related suggestion below.

File: docs/reference/ingest/processors/community-id.asciidoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+| `destination_ip`   | no       | `destination.ip` | Field containing the destination IP address.
+| `destination_port` | no       | `destination.port` | Field containing the destination port.
+| `iana_number`      | no       | `network.iana_number` | Field containing the IANA number.
+| `icmp_type`        | no       | `icmp.type`   | Field containing the ICMP type.
+| `icmp_code`        | no       | `icmp.code`   | Field containing the ICMP code.
+| `transport`        | no       | `network.transport` | Field containing the transport protocol.
+Used only when the `iana_number` field is not present.
+| `target_field`     | no       | `network.community_id` | Output field for the community ID.
+| `seed`             | no       | `0`           | Seed for the community ID hash. Must be between
+0 and 65535. This can be use to prevent hash collisions between network domains
```
Comment (jrodewig) [inline docs/reference/ingest/processors/community-id.asciidoc:28]: It's not clear if this range is inclusive or exclusive. I'd just add a parenthetical afterward, e.g. `(inclusive)`.
```suggestion
`0` and `65535`. The seed prevents hash collisions between network domains,
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left a few suggestions but nothing I would consider blocking (except a minor typo).
Thanks, @danhermann!

**Pull request merged**